### PR TITLE
SOF-1878 Blueprint configuration now maps VideoMixerType

### PR DESCRIPTION
**Pull request by LindvedKrvang**
Blueprint configuration now maps VideoMixerType.

Blueprints now supports creating TimelineObjects for TriCaster.
Note: The TriCasterTypes has been "refactored" to a hopefully more understandable format. However, the attributes of the types still _must_ match what's in TSR. If not, TSR has no idea of what to do with it. So there may be some naming that might seem weird, but to change it we would also need to change it in TSR (which we prefer not to do at the moment).

### CODE REVIEW

**Review by k-primeholding-dim — DISMISSED** [review]

**Review by RasmusAlbrektsen — DISMISSED** [review]

File: src/blueprints/tv2/helpers/tv2-studio-blueprint-configuration-mapper.ts [hunk 1]
```diff
@@ -157,6 +165,18 @@ export class Tv2StudioBlueprintConfigurationMapper {
     })
   }
 
+  private mapToVideoMixerType(videoMixerTypeString: string): VideoMixerType {
+    switch(videoMixerTypeString) {
```
Comment (KvelaGorrrrnio) [inline src/blueprints/tv2/helpers/tv2-studio-blueprint-configuration-mapper.ts]: The `String` suffix seems redundant. If you want to make it explicit that it is not of type `VideoMixerType`, then either `coreVideoMixerType` or `rawVideoMixerType` can be used :)

**Review by KvelaGorrrrnio — DISMISSED** [review]

**Comment by LindvedKrvang** [comment]
I accidentally pushed the code for SOF-1877 on to this branch, so we are going to to review SOF-1878 and SOF-1877 together (luckily SOF-1878 is small).
This means I am converting this branch to draft untill both SOF-1877 and SOF-1878 is ready for review.

File: src/blueprints/tv2/helpers/tv2-studio-blueprint-configuration-mapper.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/blueprints/tv2/helpers/tv2-studio-blueprint-configuration-mapper.ts]: Updated to `rawVideoMixerType`.

**Comment by LindvedKrvang** [comment]
Created SOF-1883 for the changing of the VideoMixerTimelineObjectFactory. That means that this PR is then ready for review.

**Review by k-primeholding-dim — DISMISSED** [review]

File: src/blueprints/tv2/timeline-object-factories/tv2-tri-caster-video-mixer-timeline-object-factory.ts [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+export class Tv2TriCasterVideoMixerTimelineObjectFactory implements Tv2VideoMixerTimelineObjectFactory {
+
+  private readonly logger: Tv2Logger
+
+  constructor(private readonly atemToTriCasterSplitScreenConverter: AtemToTriCasterSplitScreenConverter, logger: Tv2Logger) {
+    this.logger = logger.tag(Tv2TriCasterVideoMixerTimelineObjectFactory.name)
+  }
+
+
```
Comment (RasmusAlbrektsen) [inline src/blueprints/tv2/timeline-object-factories/tv2-tri-caster-video-mixer-timeline-object-factory.ts]: Extra newline here

**Review by RasmusAlbrektsen — DISMISSED** [review]

File: src/blueprints/tv2/timeline-object-factories/tv2-tri-caster-video-mixer-timeline-object-factory.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/blueprints/tv2/timeline-object-factories/tv2-tri-caster-video-mixer-timeline-object-factory.ts]: Removed

**Pull request merged**